src/com/aaronzalewski/HtmlDocumentImpl.java

```diff
@@ -65,20 +82,28 @@ private String getUrlDomain(String url) {
     }
 
     private boolean hasDocumentDomain(Element link) {
-        return domain.equals(getUrlDomain(link.attr("href"))) ||
-                ("www." + domain).equals(getUrlDomain(link.attr("href")));
+        return domain.equals(getUrlDomain(link.attr("href")));
+    }
+
+    private String cleanUrlBeginning(String url) {
+        if (url.indexOf("http://") == 0) {
+            return url;
+        }
+
+        String urlEnd = url.substring(url.lastIndexOf(domain) + domain.length());
+        return "http://" + domain + urlEnd;
     }
 
     private static class EmptyHtmlDocumentImpl implements HtmlDocument {
 
         @Override
-        public List<String> getEmailAddresses() {
-            return ImmutableList.of();
+        public Set<String> getEmailAddresses() {
+            return ImmutableSet.of();
```

Comment:
Oh sorry, I should have been more clear. Why did you choose to use an Immutable data structure?